flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopFileSystem.java

```diff
@@ -116,17 +135,19 @@ public FileStatus getFileStatus(final Path f) throws IOException {
 	}
 
 	@Override
-	public HadoopDataInputStream open(final Path f, final int bufferSize) throws IOException {
+	public FSDataInputStream open(final Path f, final int bufferSize) throws IOException {
 		final org.apache.hadoop.fs.Path path = toHadoopPath(f);
 		final org.apache.hadoop.fs.FSDataInputStream fdis = this.fs.open(path, bufferSize);
-		return new HadoopDataInputStream(fdis);
+		HadoopDataInputStream inputStream = new HadoopDataInputStream(fdis);
+		if (readBufferSize <= 0) {
```

Comment:
`readBufferSize == 0` is clearer -- it cannot be negative because of your ctor check. Also I'd add a constant for `0`. (`NO_BUFFER`, `NO_BUFFERING`, ...)